internal/services/ingest.go

```diff
@@ -389,33 +397,55 @@ func (m *ingestService) ingestProcessedData(ctx context.Context, ledgerIndexer *
 		opByID := make(map[int64]types.Operation)
 		stellarAddressesByOpID := make(map[int64]set.Set[string])
 
+		stateChanges := make([]types.StateChange, 0)
+
 		// 1. Build the data structures needed for the DB insertions
 		for participant := range indexerBuffer.Participants.Iter() {
-			if !indexerBuffer.Participants.Contains(participant) {
-				continue
-			}
-
 			// 1.1. transactions data for the DB insertions
 			participantTransactions := indexerBuffer.GetParticipantTransactions(participant)
 			for _, tx := range participantTransactions {
-				txByHash[tx.Hash] = tx
-				if _, ok := stellarAddressesByTxHash[tx.Hash]; !ok {
-					stellarAddressesByTxHash[tx.Hash] = set.NewSet[string]()
+				dataTx, err := ingestutils.ConvertTransaction(&tx)
+				if err != nil {
+					return fmt.Errorf("creating data transaction: %w", err)
+				}
+
+				txByHash[dataTx.Hash] = *dataTx
+				if _, ok := stellarAddressesByTxHash[dataTx.Hash]; !ok {
+					stellarAddressesByTxHash[dataTx.Hash] = set.NewSet[string]()
 				}
-				stellarAddressesByTxHash[tx.Hash].Add(participant)
+				stellarAddressesByTxHash[dataTx.Hash].Add(participant)
+
+				tokenTransferChanges, err := m.tokenTransferProcessor.ProcessTransaction(ctx, tx)
+				if err != nil {
+					return fmt.Errorf("processing token transfer changes: %w", err)
+				}
+
+				stateChanges = append(stateChanges, tokenTransferChanges...)
```

Comment:
You may be duplicating things here because this transaction could have been porcesed beforehand. The reason being we are looping through participants and then tx-per-participant, and often we have the same transaction connected to multiple participants.

When I designed the indexer, my intention was to process everything there (once), and then we filter things off here in the `ingestProcessedData` method.

Reply:
Ohh yeah good catch! Looks like it works for the txns and participants maps becuase they use a set so we can add duplicates without any issue. But let me resolve this